### PR TITLE
chore: add update/increase/decreaseTtl APIs

### PR DESCRIPTION
**Pull request by poppoerika**
Added update/increase/decreaseTtl APIs to the reference doc.

@NoSQLKnowHow @allenheltondev @cprice404 
I wasn't sure where this universal method belong in the doc so I added them right below the data APIs for now.
Open to suggestions and I will make updates as needed!

Ticket: https://github.com/momentohq/dev-eco-issue-tracker/issues/209

### CODE REVIEW

File: docs/develop/api-reference/index.mdx [hunk 1]
```diff
@@ -505,4 +505,38 @@ For more in-depth information on usage, see [collection data types](./../datatyp
 | Dictionary CDTs   | :white_check_mark: | :x:                | :white_check_mark: | :x:                | :white_check_mark: | :x:                | :x:                | :x:                |
 | Set CDTs          | :white_check_mark: | :x:                | :white_check_mark: | :x:                | :white_check_mark: | :x:                | :x:                | :x:                |
 | List CDTs         | :white_check_mark: | :x:                | :white_check_mark: | :x:                | :white_check_mark: | :x:                | :x:                | :x:                |
-| Flush Cache       | :white_check_mark: | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :white_check_mark: |
\ No newline at end of file
+| Flush Cache       | :white_check_mark: | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :white_check_mark: |
+
+## Universal Methods
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:510]: This section should be above the table here. I'd also rename it to `Time to Live APIs`

File: docs/develop/api-reference/index.mdx [hunk 2]
```diff
@@ -505,4 +505,38 @@ For more in-depth information on usage, see [collection data types](./../datatyp
...
+
+These APIs apply across all cache types.
+
+### UpdateTtl
+
+Sets a given Time To Live (TTL) seconds for a key to the provided value.
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:516]: This description is a little ambiguous to me. I suggest using something like this:
```
Overwrites the TTL of a cache item with the provided value in seconds.
```

File: docs/develop/api-reference/index.mdx [hunk 3]
```diff
@@ -505,4 +505,38 @@ For more in-depth information on usage, see [collection data types](./../datatyp
...
+
+| Name      | Type   | Description                                     |
+| --------- | ------ | ----------------------------------------------- |
+| cacheName | String | Name of the cache.                              |
+| key       | String \| bytes | The key under which the value's TTL is to be updated. |
+| ttl       | Duration | Time to live that you want to update in cache in seconds. |
+
+### IncreaseTtl
+
+Sets a given Time To Live (TTL) seconds for a key to the provided value only if it would increase the TTL.
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:526]: This would benefit from a couple of examples to show when it would or wouldn't update and what the values would be updated to.

File: docs/develop/api-reference/index.mdx [hunk 4]
```diff
@@ -505,4 +505,38 @@ For more in-depth information on usage, see [collection data types](./../datatyp
...
+
+| Name      | Type   | Description                                     |
+| --------- | ------ | ----------------------------------------------- |
+| cacheName | String | Name of the cache.                              |
+| key       | String \| bytes | The key under which the value's TTL is to be increased. |
+| ttl       | Duration | Time to live that you want to increase to. |
+
+### DecreaseTtl
+
+Sets a given Time To Live (TTL) seconds for a key to the provided value only if it would decrease the TTL.
```
Comment (allenheltondev) [inline docs/develop/api-reference/index.mdx:536]: This could use examples as well

File: docs/develop/api-reference/response-objects.md [hunk 5]
```diff
@@ -38,4 +38,13 @@ For fetch commands, the key or field exists in the cache and there is a "cache h
 
 ### Miss
 
-For fetch commands, the key or field does not exist and it a "cache miss."
\ No newline at end of file
+For fetch commands, the key or field does not exist and it a "cache miss."
```
Comment (allenheltondev) [inline docs/develop/api-reference/response-objects.md:41]: I'd reword this since we already have context of it being a Miss

```
* **Fetch commands** - the key or field does not exist
* **TTL commands** - the key is not present
```

After writing those out, is there actually a difference between fetch/ttl commands for a miss?

File: docs/develop/api-reference/response-objects.md [hunk 6]
```diff
@@ -38,4 +38,13 @@ For fetch commands, the key or field exists in the cache and there is a "cache h
...
+For update/increase/decreaseTtl commands, the update was not applied due to key not being present.
+
+### Set
+
+For update/increase/decreaseTtl commands, the update was applied.
```
Comment (allenheltondev) [inline docs/develop/api-reference/response-objects.md:46]: I'd reword to summarize all TTL commands

```
For TTL commands, the update was applied successfully
```

File: docs/develop/api-reference/response-objects.md [hunk 7]
```diff
@@ -38,4 +38,13 @@ For fetch commands, the key or field exists in the cache and there is a "cache h
...
+
+### NotSet
+
+For update/increase/decreaseTtl commands, the update was not applied and no change was made to the existing TTL.
```
Comment (allenheltondev) [inline docs/develop/api-reference/response-objects.md:50]: Reword this for consistency above
```
For TTL commands, the update was not applied and no change was made to the existing TTL.
```

**Review by allenheltondev — CHANGES_REQUESTED** [review]

File: docs/develop/api-reference/response-objects.md (hunk 5 above)
Comment (poppoerika) [reply docs/develop/api-reference/response-objects.md:41]: That's a good point. 
For the fetch commands, we call it a "cache miss" but for not the ttl commands.
But we don't need to phrase that way to describe the "Miss" response type, so I can just remove `For ~commands` part 👍

Comment (poppoerika) [reply docs/develop/api-reference/response-objects.md:41]: I will do the same thing for the `Hit` response type.

**Comment by poppoerika** [comment]
Closing this PR since this is included in this PR: https://github.com/momentohq/public-dev-docs/pull/131
